benchmark/webstreams/readable-read.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+const common = require('../common.js');
+const { ReadableStream } = require('node:stream/web');
+
+const bench = common.createBenchmark(main, {
+  n: [1e5],
+  type: ['normal', 'byob'],
+});
+
+async function main({ n, type }) {
+  switch (type) {
+    case 'normal': {
+      const rs = new ReadableStream({
+        pull: function(controller) {
+          controller.enqueue('a');
+        },
+      });
+      const reader = rs.getReader();
+      let x = null;
+      bench.start();
+      for (let i = 0; i < n; i++) {
+        const { value } = await reader.read();
+        x = value;
+      }
+      bench.end(n);
+      console.assert(x);
+      break;
+    }
+    case 'byob': {
+      const encode = new TextEncoder();
+      const rs = new ReadableStream({
+        type: 'bytes',
+        pull: function(controller) {
+          controller.enqueue(encode.encode('a'));
+        },
+      });
+      const reader = rs.getReader({ mode: 'byob' });
+      let x = null;
+      bench.start();
+      for (let i = 0; i < n; i++) {
+        const { value } = await reader.read(new Uint8Array(1));
```

Comment:
I think `new Uint8Array(1)` is better to be cached, otherwise, we will test the performance of Uint8Array and the `read` function.

Reply:
Ah indeed, updating

Reply:
but trying to use the same buffer again would result in error with buffer being detached from previous call no?

Reply:
get the following error on trying this
```js
const encode = new TextEncoder();
const rs = new ReadableStream({
  type: 'bytes',
  pull: function (controller) {
    controller.enqueue(encode.encode('a'));
  },
});
const reader = rs.getReader({ mode: 'byob' });
let x = null;
const buf = new Uint8Array(1);
for (let i = 0; i < 100; i++) {
  const { value } = await reader.read(buf);
  x = value;
}
console.assert(x);
```

```console
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_INVALID_STATE]: Invalid state: View or Viewed ArrayBuffer is zero-length or detached
    at new NodeError (node:internal/errors:405:5)
    at ReadableStreamBYOBReader.read (node:internal/webstreams/readablestream:953:9)
    at file:///Users/debadreechatterjee/Documents/personal/node/test.mjs:12:34 {
  code: 'ERR_INVALID_STATE'
}

Node.js v20.6.1
```

Reply:
Oh, I didn't know about that behavior, in this case, `new Uint8Array` is fine since we don't have a good alternative.